File: impl/sdk/ts/scripts/release.ts
// New release helper, mirroring resonate-sdk-py's scripts/new-release.py.
// Opens the GitHub "new release" page with the root package's version as the
// tag (vX.Y.Z). One release publishes the whole monorepo (see cd.yml).
//
// Usage: npx tsx scripts/release.ts

import { execFile } from "node:child_process";
import { readFileSync } from "node:fs";
import { resolve } from "node:path";

const REPO = "https://github.com/resonatehq/resonate-sdk-ts";

const { version } = JSON.parse(readFileSync(resolve(process.cwd(), "package.json"), "utf-8")) as { version?: string };
if (!version) {
  console.error('❌ package.json is missing "version".');
  process.exit(1);
}

const tag = `v${version}`;
const url = `${REPO}/releases/new?${new URLSearchParams({ tag, title: tag })}`;

console.log(`🔗 Opening: ${url}`);
// Native OS opener — no dependency (macOS: open, Windows: start, Linux: xdg-open).
const opener = process.platform === "darwin" ? "open" : process.platform === "win32" ? "start" : "xdg-open";
execFile(opener, [url], (err) => {
  if (err) console.warn(`⚠️  Could not launch a browser (${err.message}). Open the URL above.`);
});

File: impl/sdk/java/scripts/release.py
"""New release."""

from __future__ import annotations

import pathlib
import re
import webbrowser
from urllib.parse import urlencode

SOURCE = "https://github.com/resonatehq/resonate-sdk-java"
VERSION_RE = re.compile(r'^version\s*=\s*"([^"]+)"', re.MULTILINE)


def read_version(cwd: pathlib.Path) -> str:
    """Read `version = "x.y.z"` from build.gradle.kts."""
    text = cwd.joinpath("build.gradle.kts").read_text(encoding="utf-8")
    match = VERSION_RE.search(text)
    if not match:
        raise RuntimeError("version not found in build.gradle.kts")
    return match.group(1)


def main() -> None:
    """Prepare new release."""
    version = read_version(cwd=pathlib.Path.cwd())
    params = urlencode(query={"title": f"v{version}", "tag": f"v{version}"})
    webbrowser.open_new_tab(url=f"{SOURCE}/releases/new?{params}")


if __name__ == "__main__":
    main()
